pkg/kubelet/kubelet_pods.go

```diff
@@ -1503,13 +1503,24 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	// ensure the probe managers have up to date status for containers
 	kl.probeManager.UpdatePodStatus(pod.UID, s)
 
-	// preserve all conditions not owned by the kubelet
+	// preserve all conditions not owned nor shared by the kubelet
 	s.Conditions = make([]v1.PodCondition, 0, len(pod.Status.Conditions)+1)
 	for _, c := range pod.Status.Conditions {
-		if !kubetypes.PodConditionByKubelet(c.Type) {
+		if !kubetypes.PodConditionByKubelet(c.Type) && !kubetypes.PodConditionSharedByKubelet(c.Type) {
 			s.Conditions = append(s.Conditions, c)
```

Comment:
I'm unsure about this change... why wouldn't this follow the same pattern of preserving the condition if it already exists?

Reply:
no reason, introduced an analogous change as in mergePodStatus to simplify